telegram/utils/helpers.py

```diff
@@ -58,63 +57,32 @@ def escape_markdown(text, version=1, entity_type=None):
             ``version=2``, will be ignored else.
     """
     if int(version) == 1:
-        escape_chars = '\*_`\['
+        escape_chars = r'_*`['
     elif int(version) == 2:
         if entity_type == 'pre' or entity_type == 'code':
-            escape_chars = '`\\\\'
+            escape_chars = r'\`'
         elif entity_type == 'text_link':
-            escape_chars = ')\\\\'
+            escape_chars = r'\)'
         else:
-            escape_chars = '_*\[\]()~`>\#\+\-=|{}\.!'
+            escape_chars = r'_*[]()~`>#+-=|{}.!'
     else:
         raise ValueError('Markdown version musst be either 1 or 2!')
 
-    return re.sub(r'([%s])' % escape_chars, r'\\\1', text)
+    return re.sub('([{}])'.format(re.escape(escape_chars)), r'\\\1', text)
 
 
 # -------- date/time related helpers --------
 # TODO: add generic specification of UTC for naive datetimes to docs
 
-if hasattr(dtm, 'timezone'):
-    # Python 3.3+
-    def _datetime_to_float_timestamp(dt_obj):
-        if dt_obj.tzinfo is None:
-            dt_obj = dt_obj.replace(tzinfo=_UTC)
-        return dt_obj.timestamp()
-
-    _UtcOffsetTimezone = dtm.timezone
-    _UTC = dtm.timezone.utc
-else:
-    # Python < 3.3 (incl 2.7)
-
-    # hardcoded timezone class (`datetime.timezone` isn't available in py2)
-    class _UtcOffsetTimezone(dtm.tzinfo):
-        def __init__(self, offset):
-            self.offset = offset
-
-        def tzname(self, dt):
-            return 'UTC +{}'.format(self.offset)
-
-        def utcoffset(self, dt):
-            return self.offset
-
-        def dst(self, dt):
-            return dtm.timedelta(0)
-
-    _UTC = _UtcOffsetTimezone(dtm.timedelta(0))
-    __EPOCH_DT = dtm.datetime.fromtimestamp(0, tz=_UTC)
-    __NAIVE_EPOCH_DT = __EPOCH_DT.replace(tzinfo=None)
-
-    # _datetime_to_float_timestamp
-    # Not using future.backports.datetime here as datetime value might be an input from the user,
-    # making every isinstace() call more delicate. So we just use our own compat layer.
-    def _datetime_to_float_timestamp(dt_obj):
-        epoch_dt = __EPOCH_DT if dt_obj.tzinfo is not None else __NAIVE_EPOCH_DT
-        return (dt_obj - epoch_dt).total_seconds()
+def _datetime_to_float_timestamp(dt_obj):
```

Comment:
We already changed this in #1621, so this will give conflicts for sure … Just a heads up.